### PR TITLE
Adds processor configuration support for AWS cloudwatch metrics

### PR DESCRIPTION
**Pull request by philippkahr**
## What does this PR do?

Adds processor configuration support for AWS cloudwatch metrics

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

### CODE REVIEW

**Comment by philippkahr** [comment]
@elastic/obs-cloud-monitoring can someone take a look and review please :)?

**Comment by kaiyan-sheng** [comment]
@philippkahr Hello! Thanks for creating this PR. Do we have a use case to add processors in cloudwatch metrics? Or is there a github issue for it with more details? Thanks!!

**Comment by philippkahr** [comment]
Consistency. Nearly all integrations support the processor config, it's odd to always find a few that do not support it.

**Comment by tommyers-elastic** [comment]
seems fine to me - if we are doing it for one metrics data stream, we should probably do it for all though.

**Review by tommyers-elastic — APPROVED** [review]
LGTM

File: packages/aws/data_stream/cloudwatch_metrics/manifest.yml [hunk 1]
```diff
@@ -58,5 +58,13 @@ streams:
...
               # - key: created-by
                 # value: foo
+      - name: processors
+        type: yaml
+        title: Processors
+        multi: false
+        required: false
+        show_user: false
+        description: >
+          Processors are used to reduce the number of fields in the exported event or to enhance the event with metadata. This executes in the agent before the logs are parsed. See [Processors](https://www.elastic.co/guide/en/beats/filebeat/current/filtering-and-enhancing-data.html) for details.
```
Comment (kaiyan-sheng) [inline packages/aws/data_stream/cloudwatch_metrics/manifest.yml:68]: The description here mentioned logs parsing, we should change it for metrics.

**Pull request closed without merging**